public/app/features/dashboard-scene/pages/DashboardScenePageStateManager.ts

```diff
@@ -100,6 +100,27 @@ export class DashboardScenePageStateManager extends StateManagerBase<DashboardSc
     return rsp;
   }
 
+  public async loadSnapshot(slug: string) {
```

Comment:
These logic could be inside `loadDashboard` and `loadScene` but I didn't want to introduce additional logic related to Snapshots. I could change it if it does not make sense.